src/libs/DateUtils.ts

```diff
@@ -336,6 +453,154 @@ function getStatusUntilDate(inputDate: string): string {
+ * returns {string} - Date with updated time in "YYYY-MM-DD HH:mm:ss" format.
+ */
+const combineDateAndTime = (updatedTime: string, inputDateTime: string): string => {
+    if (!updatedTime || !inputDateTime) {
+        return '';
+    }
+
+    let parsedTime: Date | null = null;
+    if (updatedTime.includes('-')) {
+        // it's in "yyyy-MM-dd HH:mm:ss" format
+        const tempTime = parse(updatedTime, 'yyyy-MM-dd HH:mm:ss', new Date());
+        if (isValid(tempTime)) {
+            parsedTime = tempTime;
+        }
+    } else if (updatedTime.includes(':')) {
+        // it's in "hh:mm a" format
+        const tempTime = parse(updatedTime, 'hh:mm a', new Date());
+        if (isValid(tempTime)) {
+            parsedTime = tempTime;
+        }
+    }
+
+    if (!parsedTime) {
+        return '';
+    }
+
+    let parsedDateTime: Date | null = null;
+    if (inputDateTime.includes(':')) {
+        // Check if it includes time
+        const tempDateTime = parse(inputDateTime, 'yyyy-MM-dd HH:mm:ss', new Date());
+        if (isValid(tempDateTime)) {
+            parsedDateTime = tempDateTime;
+        }
+    } else {
+        const tempDateTime = parse(inputDateTime, 'yyyy-MM-dd', new Date());
+        if (isValid(tempDateTime)) {
+            parsedDateTime = tempDateTime;
+        }
+    }
+
+    if (!parsedDateTime) {
+        return '';
+    }
+
+    const updatedDateTime = set(parsedDateTime, {
+        hours: parsedTime.getHours(),
+        minutes: parsedTime.getMinutes(),
+        seconds: parsedTime.getSeconds(),
+    });
+
+    return format(updatedDateTime, 'yyyy-MM-dd HH:mm:ss');
+};
+
+/**
+ * param {String} dateTime // 'HH:mm:ss'
+ * returns {Object}
+ * example get12HourTimeObjectFromDate('11:10:00') // {hour: '11', minute: '10', period: 'AM'}
+ */
+function get12HourTimeObjectFromDate(dateTime: string): {hour: string; minute: string; period: string} {
+    if (!dateTime) {
+        return {
+            hour: '12',
+            minute: '00',
+            period: 'PM',
+        };
+    }
+    const parsedTime = parse(dateTime, 'hh:mm a', new Date());
+    return {
+        hour: format(parsedTime, 'hh'),
+        minute: format(parsedTime, 'mm'),
+        period: format(parsedTime, 'a').toUpperCase(),
+    };
+}
+
+/**
+ * param {String} timeString
+ * returns {String}
+ * example getTimePeriod('11:10 PM') // 'PM'
+ */
+function getTimePeriod(timeString: string): TimePeriod {
+    const parts = timeString.split(' ');
+    return parts[1] as TimePeriod;
+}
+
+/**
+ * param {String} dateTimeStringFirst // YYYY-MM-DD HH:mm:ss
+ * param {String} dateTimeStringSecond // YYYY-MM-DD HH:mm:ss
+ * returns {Boolean}
+ */
+function areDatesIdentical(dateTimeStringFirst: string, dateTimeStringSecond: string): boolean {
+    const date1 = parse(dateTimeStringFirst, 'yyyy-MM-dd HH:mm:ss', new Date());
+    const date2 = parse(dateTimeStringSecond, 'yyyy-MM-dd HH:mm:ss', new Date());
+
+    return isSameMinute(date1, date2);
```

Comment:
Any reason we're not using `isSameSecond` here?

Reply:
No, I mean we don't use seconds in the app, so there's no difference between comparing minutes and seconds. Maybe from a perspective of reusability, I'll change it to compare seconds. 👍🏻